### PR TITLE
Feat/sso unified signin

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { lift } from 'ramda'
+
+import { ExtractSuccess } from '@core/types'
+import { selectors } from 'data'
+import { RootState } from 'data/rootReducer'
+import { LoginFormType, LoginSteps } from 'data/types'
+
+import { LOGIN_FORM_NAME } from './model'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/selectors.ts:8]: can we use the constant from `data` so we dont have this in two places? i.e. i think its `data.auth.model.LOGIN_FORM`

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/model.tsx [hunk 2]
```diff
@@ -3,24 +3,62 @@ import { FormattedMessage } from 'react-intl'
...
+  SpinningLoader,
+  Text,
+  TextGroup
+} from 'blockchain-info-components'
 import { FormLabel } from 'components/Form'
+import { Wrapper } from 'components/Public'
 import { LoginFormType, LoginSteps } from 'data/types'
 import { media } from 'services/styles'
 
 export const LOGIN_FORM_NAME = 'login'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/model.tsx:20]: made a comment below about this, but lets remove this and use the one in data.auth.model

Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Login/model.tsx:20]: you could actually just import from data here and then re-export for this model file so you dont have to track down all the other spots and update

**Review by schnogz — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -148,6 +148,7 @@
     "cross-env": "7.0.3",
     "css-loader": "4.3.0",
     "cz-conventional-changelog": "3.3.0",
+    "base64url": "^3.0.1",
```
Comment (schnogz) [inline package.json:151]: remove the `^` and then re-run yarn to get new lockfile

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/balance/sagas.js [hunk 4]
```diff
@@ -7,6 +7,16 @@ import { actions, actionTypes, selectors } from 'data'
...
+export const fetchBalances = function* () {
+  yield put(actions.core.data.bch.fetchData())
+  yield put(actions.core.data.btc.fetchData())
+  yield put(actions.core.data.xlm.fetchData())
+  yield put(actions.core.data.eth.fetchData())
+  yield put(actions.core.data.eth.fetchErc20Data())
+  yield put(actions.components.refresh.refreshRates())
+  yield put(actions.custodial.fetchRecentSwapTxs())
+}
+
```
[the remark this answers is not in the document]
Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/data/balance/sagas.js:19]: Oof, I don't know. Probably merge conflict mistake will fix.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/todoList.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+1. clean up selectors in index.tsx
+   - single formValues import
+   - move everything to selectors.ts
+2. cleanup continueLoginProcess function
+3. Login/AccountUnification/UpgradeSuccess probably only needs one template
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/todoList.md:5]: can probably delete this file

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/Wallet/TwoFA/index.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          >
+            {submitting || busy ? (
+              <HeartbeatLoader height='20px' width='20px' color='white' />
+            ) : (
+              <Text color='whiteFade900' size='16px' weight={600}>
+                <FormattedMessage id='scenes.login.login' defaultMessage='Log In' />
+              </Text>
+            )}
+          </ActionButton>
+          <WalletNeedHelpLink authActions={props.authActions} origin='PASSWORD' />
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/Wallet/TwoFA/index.tsx:139]: should origin be `2FA`?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/Exchange/TwoFA/index.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+          >
+            {submitting ? (
+              <HeartbeatLoader height='20px' width='20px' color='white' />
+            ) : (
+              <Text color='whiteFade900' size='16px' weight={600}>
+                <FormattedMessage id='buttons.continue' defaultMessage='Continue' />
+              </Text>
+            )}
+          </ActionButton>
+          <ExchangeNeedHelpLink authActions={authActions} origin='IDENTIFIER' />
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/Exchange/TwoFA/index.tsx:87]: is origin correct?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/AccountUnification/UpgradePassword/index.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        >
+          {/* {submitting ? (
+          <HeartbeatLoader height='20px' width='20px' color='white' />
+        ) : ( */}
+          <Text color='whiteFade900' size='16px' weight={600}>
+            <FormattedMessage id='buttons.upgrade_account' defaultMessage='Upgrade Account' />
+          </Text>
+          {/* )} */}
+        </ActionButton>
+        {/* <NeedHelpLink authActions={props.authActions} origin='UPGRADE_ACCOUNT_NEW_PASSWORD' /> */}
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/AccountUnification/UpgradePassword/index.tsx:68]: are these commented out lines needed? if not, lets remove

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/AccountUnification/ProductPicker/index.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+  position: relative;
+  cursor: pointer;
+  border-top: ${(props) => `1px solid ${props.theme.grey100}`};
+  &:first-child {
+    border-top: 0;
+  }
+`
+
+// TODO add media tag for
+// min width
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/AccountUnification/ProductPicker/index.tsx:23]: this needed? can we implement?

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.utils.ts [hunk 10]
```diff
@@ -1,43 +1,175 @@
...
+          // send them to exchange password screen
+          yield put(
+            actions.auth.setAccountUnificationFlowType(AccountUnificationFlows.EXCHANGE_UPGRADE)
+          )
+        }
+      } else if (unified) {
+        actions.auth.setAccountUnificationFlowType(AccountUnificationFlows.UNIFIED)
+      }
+    }
+    // TODDO: WRAP ABOVE IN FEATURE FLAG
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.utils.ts:74]: this needs to be done before merge?

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/assets/locales/index.d.ts [hunk 11]
```diff
@@ -2288,25 +2297,41 @@ type MessagesType = {
...
   'scenes.login.wallet.connected.description_1': 'We sent your connected mobile device a notification. Open the app to confirm auto-log in on the web.'
   'scenes.login.wallet.connected.description_2': 'Didn’t get the notification? Make sure you have push notifications enabled. You can also scan this QR with your mobile app to login.'
-  'scenes.login.wallet.mobile_login.title': 'Log in with mobile app'
+  'scenes.login.wallet.mobile_app_login.title': 'Or Log in with Mobile App'
   'scenes.login.wallet.mobile_login.description_one': 'Scan this QR code with the Blockchain.com mobile app.'
-  'scenes.login.wallet.mobile_login.description.ios': 'iOS - Tap the Menu button at the top left corner of the app to reveal Web Log In option.'
-  'scenes.login.wallet.mobile_login.description.android': 'Android - Tap the QR code icon at the top right corner of the app.'
+  'scenes.login.wallet.mobile_login.description.ios': '<b>iOS</b> - Tap the Menu button at the top left corner of the app to reveal Web Log In option.'
+  'scenes.login.wallet.mobile_login.description.android': '<b>Android</b> - Tap the QR code icon at the top right corner of the app.'
```
[the remark this answers is not in the document]
Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/assets/locales/index.d.ts:2334]: I think the instructions are different for the different platforms. Could add an icon or an 'OR'. Will follow up about this with design.

**Pull request merged**